elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/HibernateTransaction.java

```diff
@@ -248,6 +249,20 @@ public void commit() {
         return new ScrollableIterator(sessionCriteria.scroll(scrollMode));
     }
 
+    @Override
+    public <T> Long getTotalRecords(Class<T> entityClass, FilterScope filterScope) {
+        final Criterion criterion = filterScope.getCriterion(NOT, AND, OR);
+        final CriteriaExplorer criteriaExplorer = new CriteriaExplorer(
+                entityClass, filterScope.getRequestScope(), criterion);
+        final Criteria sessionCriteria = session.createCriteria(entityClass);
+
+        criteriaExplorer.buildCriteria(sessionCriteria, session);
+
+        sessionCriteria.setProjection(Projections.rowCount());
```

Comment:
This is a thorny problem, because `total_num_records_filtered` depends on evaluating permissions on EVERY record in SQL. This sounds like something that should be done OOB after the first query. 

The first query would say that this is page 1 of ?? pages and we would kick off a background job that fetches all of the objects from the database. After we evaluate permissions on all of the returned objects and stores the count into a map keyed on user and object requested so that on subsequent requests we can say this is page 2 of 8.